src/core/dir.c

```diff
@@ -103,56 +52,41 @@ static int compare_lower(const void *va, const void *vb)
     return string_compare_case_insensitive(*(const char**)va, *(const char**)vb);
 }
 
+static int add_file_with_extension_to_listing(const char *filename)
+{
+    if (data.listing.num_files >= data.max_files) {
+        expand_dir_listing();
+    }
+    strncpy(data.listing.files[data.listing.num_files], filename, FILE_NAME_MAX);
+    data.listing.files[data.listing.num_files][FILE_NAME_MAX - 1] = 0;
+    ++data.listing.num_files;
+    return LIST_NO_MATCH;
```

Comment:
It feels weird to return `LIST_NO_MATCH` for getting a list of files. Perhaps also define a constant `LIST_CONTINUE` with the same value as `LIST_NO_MATCH`?